### PR TITLE
add test of karmada-scheduler-estimator

### PR DESCRIPTION
**Pull request by Garrybest**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Step 8 of #617.

**Which issue(s) this PR fixes**:
Fixes part of #617

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/helper/resource.go [hunk 1]
```diff
@@ -242,3 +253,129 @@ func NewJob(namespace string, name string) *batchv1.Job {
 		},
 	}
 }
+
```
Comment (XiShanYongYe-Chang) [inline test/helper/resource.go:256]: These methods are used only in UT, should they be placed in the corresponding `pkg/estimator/server` package?

File: pkg/estimator/server/server.go [hunk 2]
```diff
@@ -120,9 +118,6 @@ func (es *AccurateSchedulerEstimatorServer) Start(ctx context.Context) error {
 	go func() {
 		<-stopCh
 		s.GracefulStop()
-		if err := es.httpServer.Shutdown(context.Background()); nil != err {
```
[the remark this answers is not in the document]
Comment (Garrybest) [reply pkg/estimator/server/server.go:123]: Well, I have made a duplicate code here in #700. The http server has not been used. So I remove it from the file for adapting unit test.

File: test/helper/resource.go (hunk 1 above)
Comment (Garrybest) [reply test/helper/resource.go:256]: I think these code will soon be used in the future test, like #697.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve
Nice work!

**Pull request merged**